zellij-utils/src/channels.rs

```diff
@@ -2,28 +2,18 @@
 
 use async_std::task_local;
 use std::cell::RefCell;
-use std::sync::mpsc;
 
 use crate::errors::{get_current_ctx, ErrorContext};
+pub use crossbeam::channel::{bounded, unbounded, Receiver, RecvError, Select, SendError, Sender};
 
 /// An [MPSC](mpsc) asynchronous channel with added error context.
-pub type ChannelWithContext<T> = (
-    mpsc::Sender<(T, ErrorContext)>,
-    mpsc::Receiver<(T, ErrorContext)>,
-);
-/// An [MPSC](mpsc) synchronous channel with added error context.
-pub type SyncChannelWithContext<T> = (
-    mpsc::SyncSender<(T, ErrorContext)>,
-    mpsc::Receiver<(T, ErrorContext)>,
-);
+pub type ChannelWithContext<T> = (Sender<(T, ErrorContext)>, Receiver<(T, ErrorContext)>);
 
 /// Wrappers around the two standard [MPSC](mpsc) sender types, [`mpsc::Sender`] and [`mpsc::SyncSender`], with an additional [`ErrorContext`].
 #[derive(Clone)]
 pub enum SenderType<T: Clone> {
```

Comment:
I think we can remove the `SenderType` enum entirely since we only have a single type of sender. We can have the `crossbeam_channel::Sender` directly as a field of `SenderWithContext`.

Reply:
That's right, although I liked the idea of abstracting away the queue implementation in the original code, even though it was leaky quite.

I'll remove it in a subsequent patch.

Reply:
> even though it was leaky quite.

I'm sorry I didn't quite understand what you meant by that. What's wrong with it?

Reply:
That was a typo, I meant "quite leaky". If you are not familiar with the phrase "leaky abstraction", it means that what was being abstracted was still visible by the users of the abstractions, in this case mpsc types.

But it was all right, crossbeam is almost a drop-in replacement, so I barely had to do anything to change it.

Reply:
Ohh. TIL. Thanks for clarifying that 🙂. I wasn't familiar with the phrase.